libarchive/archive_read_support_compression_program.c

```diff
@@ -156,6 +156,8 @@ archive_read_support_compression_program_signature(struct archive *_a,
 	if (state == NULL)
 		return (ARCHIVE_FATAL);
```

Comment:
Can we make this 'goto memerr' too?  I think that more closely matches the upstream anyway?